typing-element.js

```diff
@@ -1,19 +1,47 @@
 import AnimationElement from "./animation-element.js";
 
+/**
+ * Checks for HTML tags
+ * @property {function} isIncompleteHTMLTag
+ * @param {string} text
+ * @returns {boolean} True for incomplete HTML tag
+ *
+ */
 function isIncompleteHTMLTag(text){
     return /<\/*$/gmi.test(text) || /(<\w*(?:\s+\w+=\"[^"]+\")*)(?=[^>]+(?:<|$))/gmi.test(text)
 }
+
+
+/**
+ * Checks character type
+ * @param {string} text 
+ * @returns {boolean} True for invisible chararacter, false for others
+ */
 function isInvisibleChar(text){
     return /\s+$/gmi.test(text)
 }
 
+/**
+ * Class to create a Typing Element
+ */
+
 class TypingElement extends AnimationElement{
+    
+    /**
+     * @type {string}
+     */
     typedLetters;
     #slot;
     #span;
     get text(){
         return this.#slot.assignedElements()[0].innerHTML;
     }
+
+    /** 
+     * Creates a TypingElement object
+     * @property {function} constructor
+     * @returns {object} New TypingElement Object
```

Comment:
I think this one can be used with \@constructor, or even ommited. I don't know if returning object can mess up with VSCode Intellisense